### PR TITLE
Displaying conflicts in diff3 style

### PR DESCRIPTION
**Pull request by jvcoutinho**
This PR solves #49 and #62.

- Conflicts generated by S3M are now displayed in Diff3 style:
```
<<<<<<< MINE
||||||| BASE
>>>>>>> YOURS
```

Topics to address:

- [x]  JGit doesn't support Diff3 style. An [issue](https://bugs.eclipse.org/bugs/show_bug.cgi?id=442284) was opened there 5 years ago, but no one implemented it. This means textual merge isn't affected by these changes currently. Should we implement it?

- [x] These changes are not parameterized (the base is displayed by default). Should I parameterize?

### CODE REVIEW

**Comment by guilhermejccavalcanti** [comment]
@jvcoutinho as you mentioned "JGit doesn't support Diff3 style.". So how do you fill the base content of the merge conflict? (I haven't look the code yet).

**Comment by jvcoutinho** [comment]
@guilhermejccavalcanti I've added a new constructor for MergeConflict.
https://github.com/guilhermejccavalcanti/jFSTMerge/blob/6f92aa4736e54844bf3fc6055493a8dd8dffd412/src/main/java/br/ufpe/cin/mergers/util/MergeConflict.java#L26-L37

Of course, this means only calls for this constructor can display conflicts in this format (the handlers).

**Comment by guilhermejccavalcanti** [comment]
Ok, so this PR only partially address the issue, right?
For instance, to merge method bodies, we call JGit textual merge, but JGit does not show the base content of the conflict...

**Comment by jvcoutinho** [comment]
@guilhermejccavalcanti Yes. I'll address the rest when we decide if we should implement this function on JGit or simply manipulate the result at hand.

**Comment by guilhermejccavalcanti** [comment]
@jvcoutinho tu verificou esse erro no Travis?

**Comment by jvcoutinho** [comment]
@guilhermejccavalcanti tudo certo agora

File: src/test/java/br/ufpe/cin/mergers/handlers/renaming/SafeRenamingHandlerTest.java [hunk 1]
```diff
@@ -183,7 +183,7 @@ public void testHandle_whenBothDeleteMethod_shouldNotReportConflict() {
 	public void testHandle_abstractMethods_whenBothRenameToDifferentName_shouldReportConflict() {
 		merge(abstractMethod1, abstractMethod2, abstractMethod3);
 		TestUtils.verifyMergeResultWithRenamingConflict(mergeContext,
-			"<<<<<<<MINEpublicabstractvoidn1();=======publicabstractvoidn2();>>>>>>>YOURS");
+			"<<<<<<<MINEpublicabstractvoidn1();|||||||BASEpublicabstractvoidm();=======publicabstractvoidn2();>>>>>>>YOURS");
 	}
```
Comment (guilhermejccavalcanti) [inline src/test/java/br/ufpe/cin/mergers/handlers/renaming/SafeRenamingHandlerTest.java]: @jvcoutinho todo conflito de handler vai exibir a base também sempre?
Pois no git isso é um parametro, por default não exibe. Deveria ser assim no nosso também.

Comment (jvcoutinho) [reply src/test/java/br/ufpe/cin/mergers/handlers/renaming/SafeRenamingHandlerTest.java]: Ok.
Por default o nosso não deveria exibir ou deveria?

**Comment by guilhermejccavalcanti** [comment]
Isso. O default seria não exibir a base.
Acho que --show-base seria um bom parâmetro.

Em qui, 25 de jul de 2019 às 12:36, João Victor <notifications@github.com>
escreveu:

**Comment by pauloborba** [comment]
> @guilhermejccavalcanti Yes. I'll address the rest when we decide if we should implement this feature on JGit or simply manipulate the result at hand.

@jvcoutinho what's the status here?

**Comment by jvcoutinho** [comment]
I've implemented the changes on JGit, but I'm struggling to push them.

Em seg, 29 de jul de 2019 10:19, Paulo Borba <notifications@github.com>
escreveu:

**Comment by jvcoutinho** [comment]
@pauloborba Ok, I could finally push the changes in JGit.
[Here](https://git.eclipse.org/r/#/c/146746/) they are waiting for a review.

**Comment by pauloborba** [comment]
@jvcoutinho check whether you followed jgit contribution rules, otherwise they won't accept your contribution.

File: src/main/java/br/ufpe/cin/mergers/TextualMerge.java [hunk 2]
```diff
@@ -10,18 +11,25 @@
...
 
+import br.ufpe.cin.app.JFSTMerge;
 import br.ufpe.cin.exceptions.ExceptionUtils;
 import br.ufpe.cin.exceptions.TextualMergeException;
+import br.ufpe.cin.files.FilesEncoding;
 import br.ufpe.cin.files.FilesManager;
 
 /**
  * Represents unstructured, linebased, textual merge.
+ * We use a temporary, still in evaluation process JGit's artifact. This build allows
```
Comment (guilhermejccavalcanti) [inline src/main/java/br/ufpe/cin/mergers/TextualMerge.java]: @jvcoutinho 
"We use a temporary JGit artifact." apenas.

**Comment by guilhermejccavalcanti** [comment]
@jvcoutinho  verifica esses conflitos pra que eu possa integrar, por favor

**Pull request merged**